### ISSUE TITLE
How do I handle the attributes,when I use SUEntitiesFill to copy face？

### ISSUE THREAD

**Issue by LItterBoy-GB** [SketchUp/api-issue-tracker#694]
I want to copy the group from ‘a.skp’ to ‘b.skp’.
I can only find the corresponding face through points and copy attributes.
Is there a simpler way？

**Comment by thomthom** [comment]
There isn't any way to add attributes via `SUGeometryInput` in the existing API. I'm logging this as a feature request.

**Comment by DanRathbun** [comment]
Another forum topic ref (27 SEP 2021): [SUEntityGetAttributeDictionary function does not work with SUGeometryInputRef
](https://forums.sketchup.com/t/suentitygetattributedictionary-function-does-not-work-with-sugeometryinputref/176649)

**Comment by DanRathbun** [comment]
The objects within a [`SUGeometryInput`](https://extensions.sketchup.com/developers/sketchup_c_api/sketchup/struct_s_u_geometry_input_ref.html) collection are not yet [`SUEntityRef`](https://extensions.sketchup.com/developers/sketchup_c_api/sketchup/struct_s_u_entity_ref.html) subclass objects, so they must be accessed by index.

**So the functions would need to be something like:**

```c
SU_RESULT SUGeometryInputSetAttribute (
  SUGeometryInputRef geom_input, SURefType object_type, size_t index,
  const char *dict_name, const char* key, SUTypedValueRef value
);
// Perhaps passing NULL as the value could delete the attribute ?

SU_RESULT SUGeometryInputGetAttribute (
  SUGeometryInputRef geom_input, SURefType object_type, size_t index,
  const char *dict_name, const char* key, SUTypedValueRef *value
);

SU_RESULT SUGeometryInputGetNumAttributes (
  SUGeometryInputRef geom_input, SURefType object_type, size_t index,
  const char *dict_name, const char* key, size_t *count
);

SU_RESULT SUGeometryInputGetAttributeKeys (
  SUGeometryInputRef geom_input, SURefType object_type, size_t index,
  size_t len, const char* keys[], size_t *count
);
```

**Comment by thomthom** [comment]
Except attributes can be nested, so the scenario quickly becomes complicated.

I wonder if it would be possible to instead create an alternative "fill" method that could return some mapping datastructure that let you map the input index with the created face. That might be a more flexible and generic approach.

**Comment by DanRathbun** [comment]
> Except attributes can be nested, so the scenario quickly becomes complicated.

True. Perhaps the `dict_name` string could accept dictionary pathnames like file path names ?
